pkg/lightning/restore/restore.go

```diff
@@ -2292,7 +2303,7 @@ func (cr *chunkRestore) deliverLoop(
 			saveCheckpoint(rc, t, engineID, cr.chunk)
 		}
 
-		func() {
+		err = func() error {
```

Comment:
If time is enough, I think we might should make a refactor on `restoreEngine`. (Not for now, yes.)

IMO, Using control flow like this seems a little weird...